### PR TITLE
Add possibility to use ModelConverters

### PR DESCRIPTION
**Pull request by hiddewie**
This enables the use of `ModelConverter`s from the plugin.

Use case: a complex model which is represented as a simple type when used as a request parameter.

### CODE REVIEW

File: src/test/resources/expectedOutput/recursive/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            application/hal+json: {}
+  /api/accounts/get:
+    get:
+      operationId: convert
+      parameters:
+      - name: parameter
+        in: query
+        schema:
+          type: string
+          format: date
```
Comment (hiddewie) [inline src/test/resources/expectedOutput/recursive/swagger.yaml:23]: This is the only place in the tests that executes the model converter.

**Comment by hiddewie** [comment]
Pull request has been updated for the new master commits

**Comment by langecode** [comment]
Swagger is actually using the Java `ServiceLoader` mechanism to discover different things, e.g., `ModelConverter` implementations. The missing support for this in the kongchen plugin was actually the primary reason for starting this plugin.

This implies that if you place a file in `META-INF/services` with the name `io.swagger.v3.core.converter.ModelConverter` which contains the fully qualified name of your model converter, then Swagger should pick this up and activate the model converter.

This also means that you can actually package model converter implementations inside separate jar files and re-use these across projects.

You can see the code for discovery in [ModelConverters](https://github.com/swagger-api/swagger-core/blob/5ed457e7698c2d22cffc806cc1a1465c816ea3bf/modules/swagger-core/src/main/java/io/swagger/v3/core/converter/ModelConverters.java#L141)

**Comment by hiddewie** [comment]
@langecode Thanks for noticing that. I have been looking through the source code of Swagger Core, but did not find that ServiceLoader call. That seems to solve my issue. I will close this PR.

**Pull request closed without merging**